### PR TITLE
Improvements to test splitting documentation

### PR DESCRIPTION
**Pull request by kbatuigas**
# Description
A brief description of the changes.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/parallelism-faster-jobs.md [hunk 1]
```diff
@@ -154,7 +154,35 @@ jobs:
...
+```shell
+circleci tests split < /path/to/items/to/split
+```
+
+* Or pipe a glob of test files.
+```shell
+circleci tests glob "test/**/*.java" | circleci tests split
+```
+
+#### b. Split by timing data
```
Comment (devinmarieb) [inline jekyll/_cci2/parallelism-faster-jobs.md:184]: Should we provide an example in this section using the `--split-by=timings` flag since there are examples in the Name and File size sections?

File: jekyll/_cci2/parallelism-faster-jobs.md [hunk 2]
```diff
@@ -209,16 +240,19 @@ The CLI attempts to autodetect whether to split by filename or classname based o
 cat my_java_test_classnames | circleci tests split --split-by=timings --timings-type=classname
 ```
 
-For tests with missing timing data, a random small timing value is assigned. You can override this default value with the `--time-default` flag:
+##### Set the default value for missing timing data
+{: #set-the-default-value-for-missing-timing-data }
+
+For partially found test results, any tests with missing data is assigned a random small value. You can override this default value with the `--time-default` flag:
```
Comment (devinmarieb) [inline jekyll/_cci2/parallelism-faster-jobs.md]: > any tests with missing data is

This needs to be either:
- ...any tests with missing data are...
or
- ...any test with missing data is...

File: jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+circleci tests glob "**/*.go" | circleci tests split --split-by=timings
+```
+
+WARNING: If you do not use `store_test_results`, there will be no timing data available to split your tests.
+
+On each successful run of a test suite, CircleCI saves timings data from the directory specified by the path in the xref:configuration-reference#storetestresults[`store_test_results` step]. This timings data consists of how long each test took to complete per filename or classname.
+
+The available timings data will then be analyzed and your tests will be split across your parallel-running containers as evenly as possible.
+
+NOTE: If no timing data is found, you will receive a message: `Error autodetecting timing type, falling back to weighting by name.`. The tests will then be split alphabetically by test name.
```
Comment (devinmarieb) [inline jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc:110]: I'm not sure the specifics around the context, but it seems we use both "timings" and "timing" throughout the pages for this. Is the context specific on when it should be pluralized or not? I'm not sure why sometimes it's pluralized and sometimes it's not. The pluralization version just sounds off to me. For example the section is called "Split by timing data" but the content says "timings data" in a few places.

File: jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+circleci tests glob "test/**/*.rb" | circleci tests split > /tmp/tests-to-run
+bundle exec rspec $(cat /tmp/tests-to-run)
+```
+
+The contents of the file `/tmp/tests-to-run` will be different in each container, based on `$CIRCLE_NODE_INDEX` and `$CIRCLE_NODE_TOTAL`.
+
+[#see-also]
+== See also
+
+* link:/parallelism-faster-jobs[Test splitting and parallelism]
```
Comment (devinmarieb) [inline jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc]: Is there a reason this is formatted as a link and not an xref?

File: jekyll/_cci2/parallelism-faster-jobs.md [hunk 5]
```diff
@@ -27,47 +27,7 @@ CircleCI test splitting lets you intelligently define where splits happen across
...
-    parallelism: 10
-    resource_class: large
-    working_directory: ~/my-app
-    steps:
-      - run: go test -v $(go list ./... | circleci tests split --split-by=timings)
-```
-
-The first time the tests are run there will be no timing data for the command to use, but on subsequent runs the test time will be optimized.
-{: class="alert alert-info"}
+It is also possible to use the CLI to [manually allocate tests](#manual-allocation) across parallel environments. Another alternative is to use [environment variables](#using-environment-variables-to-split-tests) instead of the CLI to configure split tests.
```
Comment (devinmarieb) [inline jekyll/_cci2/parallelism-faster-jobs.md:30]: Is this an internal page link reference? I don't think the #use-environment-variables-to-split-tests section exists on this page anymore.

File: jekyll/_cci2/parallelism-faster-jobs.md [hunk 6]
```diff
@@ -160,6 +160,7 @@ Test splitting by timing does not work on Windows resource classes at this time.
 ## Next steps
 {: #next-steps }
 
+* How to [use the CircleCI CLI to split tests](/docs/use-the-circleci-cli-to-split-tests)
```
Comment (devinmarieb) [inline jekyll/_cci2/parallelism-faster-jobs.md:163]: If you're going to use the new link format, can you format the entire Next Steps section to all be the same?

**Review by devinmarieb — CHANGES_REQUESTED** [review]

File: jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc (hunk 4 above)
Comment (kbatuigas) [reply jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc]: @devinmarieb hm, I think it was because I couldn't get it to work for the .md file. Let me try again

File: jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc (hunk 3 above)
Comment (kbatuigas) [reply jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc:110]: @devinmarieb I just checked a few blog posts and looks like we use "timing data" -- I'll standardize over here too

**Review by devinmarieb — APPROVED** [review]

**Review by otremblay — APPROVED** [review]
